uname_os.sh

```diff
@@ -21,6 +21,20 @@ uname_os() {
     mingw*) os="windows" ;;
   esac
 
+  # Sun Solaris and derived OS (Illumos, Oracle Solaris) reports to be the very ancient SunOS via uname not what it actually is
+  if [ "$os" = "sunos" ]; then
+    # GO 1.13 will introduce illumos as newly recognized GOOS
+    if ! grep -q illumos /etc/release; then
+      if go version | grep -q "1.13"; then
```

Comment:
@Toasterson Can you explain why you are adding a dependency on go here?  Does Illumos always have go pre-installed?  Also, what about go 1.14+ -- presumably that would also return "illumos," but I don't think the code captures that as-is?  Moreover, why do you want this value to match $GOOS, when other systems might still use "solaris"?  (I am not a sunos user so I may be missing context)

Reply:
Unfortunately there is no other way to report the correctly report GOOS on illumos based on future systems. Your library is used in the Project https://github.com/goreleaser/goreleaser to generate shell scripts to correctly evaluate GOOS variable.  And unless GOOS (OS) variable is known in your function your library call will fail. Producing a message to come here to report bugs. Thus I assume you are linked to goreleaser. In any case this library is used exclusively by goreleaser in Illumos/Solaris and nobody else thus a dependency on go is not a problem.

Reply:
I'm not loving the reliance on the `go` being installed here. I personally run `goreleaser` as a docker container preinstalled to `/bin` or equivalent on alpine - So using this project with `goreleaser` still doesn't provide confidence in Go being available.

Even fetching a remote binary (which we totally should not do, because security) would be an improvement over this.. given it's mostly chance and requires understanding the prerequisite to function.

I'm not sure what a better implementation would look like either though.. what is Go doing to determine `illumos`?

Reply:
@syntaqx if running on Alpine Linux, ` "$os" = "sunos" ]` will be false, so this is irrelevant.

But anyway this shell script should not rely on Go.

Reply:
There is an alternative implementation here:
https://github.com/golangci/golangci-lint/blob/bcd015e30713370160740e4e0adeb66bc0e988de/install.sh#L206

```sh
    sunos) [ $(uname -o) == "illumos" ] && os=illumos ;;
```

Reply:
Hi, there have been many changes in the time this PR was opened and one of them is, that we have now `uname -o` and that we can safely assume that we do no longer need to set GOOS to the one golang reports. Originally there was an overlap phase that needed that but that is no longer the case.